request.js

```diff
@@ -1413,6 +1413,28 @@ Request.prototype.jar = function (jar) {
   return self
 }
 
+Request.prototype.cloneJar = function (cookieJar) {
+  var self = this
+  var newJar, cookieString, cookieStore, domains, domainPaths, cookieUrl
+
+  newJar = self.jar()
+  cookieStore = cookieJar._jar.store.idx
```

Comment:
Yup, so basically the problem is that you can't just pass a `request` jar's store and create a new tough-cookie Cookie Jar, because now it lacks of the methods that `request` requires. For example, `request` adds a `getCookieString() ` method which is really using `tough-cookie`'s `getCookieStringSync()`, which will cause an error at some point in tough-cookie's end.

My PR is building the cookie jar correctly, I agree though I should make sure that the way I'm **retieving** the cookies is the best way. tough-cookie's public API's documentation doesn't look